### DISCUSSION TITLE
does NVchad by default support linting and debugging, for Frontend Javascript, HTML, css web development

### DISCUSSION THREAD

**Discussion by sureshsaragadam** [NvChad/NvChad#2012] [General]
Does the NvChad default installation comes with linting support for 'JAVASCRIPT' , i wish to use eslint, than tsserver, 
i want to do a basic JavaScript web development programming,

Does the default  NvChad installation works out of the box for Javascript?  do i need to extend the LSP support for javascript?  
If so do i need null-ls plugin to be installed, any other plugins need to work with Frontend Web developement with javascript.

I am wanted linting and debugging support for Javascritpt, which plugins are responsible for the same, Any other plugins need to be installed to work with "HTML" "css" "javascript".

Thankyou.

**Comment by siduck** [comment]
hi @sureshsaragadam , not by default. It only has config for lua. But you can extend it 

- Linting, autocompletion = setup lsps => tsserver, css, html, eslint 
- formatter -> setup prettier in your null-ls config 
- Debugging -> setup nvim-dap with dap-ui plugin

 Check our example_config repo, it does have web dev stuff setup ( except for debugging part ) . Also if you're new to nvchad, then please read our docs at nvchad.com, they cover everything so you wouldnt have to ask here

**Reply by sureshsaragadam** [reply]
Sure i will check the docs 

thank you.
